apps/email/js/cards/base_card.js

```diff
@@ -1,20 +1,24 @@
 'use strict';
+/* global mozIntl */
 define(function(require) {
   var mozL10n = require('l10n!'),
       base = require('./base');
 
   // Set up the global time updates for all nodes.
   (function() {
-    var formatter = new mozL10n.DateTimeFormat();
     var updatePrettyDate = function updatePrettyDate() {
-      var labels = document.querySelectorAll('[data-time]');
-      var i = labels.length;
-      while (i--) {
-        labels[i].textContent = formatter.fromNow(
-          labels[i].dataset.time,
+      var labels = [...document.querySelectorAll('[data-time]')];
+
+      labels.forEach(label => {
+        mozIntl._gaia.relativeDate(
```

Comment:
It seems like relativeDate creates a new RelativeTimeFormat instance each time it is invoked.  This differs from the code you're modifying here which creates a single formatter and then reuses it.  Based on current constants, we'd expect about 42 of these, but as we move to conversations support, that number could increase to 84 or significantly higher depending on adjustments to constants.  Is it possible to do something more batchy than this?  The "update relative timestamps" idiom is, for all apps, always going to want to be a batch operation anyways.  (Although applications may want to continue to be the ones that define what specific batch of items should be updated, and when, because of superior awareness of visibility, etc.  Although an opt-in l10n helper that did its own setInterval thing informed by a visibilitychange listener could be cool.)